Sources/SwiftLayout/Core/Activator/Activator.swift

```diff
@@ -101,17 +97,17 @@ private extension Activator {
     }
 
     static func updateConstraints(constraints: [NSLayoutConstraint]) {
-        let news = Set(constraints.weakens)
+        let news = Set(ofWeakConstraintsFrom: constraints)
         NSLayoutConstraint.activate(news.compactMap(\.origin))
     }
 
-    static func layoutIfNeeded(_ viewInfos: [ViewInformation], _ prevInfos: [ViewInformation: Set<WeakConstraint>] = [:]) {
+    static func layoutIfNeeded(_ viewInfos: [ViewInformation], _ prevInfoHashValues: Set<Int> = []) {
         for viewInfo in viewInfos {
             if viewInfo.superview == nil, let view = viewInfo.view {
                 view.setNeedsLayout()
                 view.layoutIfNeeded()
             }
-            if prevInfos[viewInfo] == nil {
+            if prevInfoHashValues.contains(viewInfo.hashValue) {
```

Comment:
```suggestion
            if !prevInfoHashValues.contains(viewInfo.hashValue) {
```
is this what you originally intended to do?

Reply:
> is this what you originally intended to do? @gmlwhdtjd

OMG!!
I wrote the wrong condition. What you're saying is the way I originally intended it.

I will fix it
Thank you!

Reply:
@oozoofrog 
Correction done.